### PR TITLE
Psalm integration

### LINKED ISSUE #11: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `psalm.xml.dist` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by ghostwriter**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Resolves #11

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -0,0 +1,5177 @@
+{
```
Comment (Ocramius) [inline composer.lock]: This `composer.lock` needs to be re-generated to a version compatible with all tested php versions

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.lock (hunk 1 above)
Comment (PowerKiKi) [reply composer.lock]: See https://github.com/laminas/laminas-skeleton-installer/pull/26 for an attempt to add composer.lock

Comment (ghostwriter) [reply composer.lock]: @Ocramius updated the lock to `PHP 7.3`. 

`phpcs` is failure should be addressed in a separate PR.

Comment (PowerKiKi) [reply composer.lock]: I'd suggest merging #26 before this one, because it restore CI entirely, including phpcs.

**Comment by Ocramius** [comment]
@samsonasik do you have time/will to rebase this one, before we release `0.4.0`?

**Comment by samsonasik** [comment]
@Ocramius I have

**Comment by samsonasik** [comment]
@Ocramius rebased.

File: composer.json [hunk 2]
```diff
@@ -25,15 +25,17 @@
     "require": {
         "php": "^7.3 || ~8.0.0 || ~8.1.0",
         "composer-plugin-api": "^1.0 || ^2.0",
-        "laminas/laminas-component-installer": "^1.0 || ^2.1.2"
+        "laminas/laminas-component-installer": "^1.0 || ^2.1.2",
+        "vimeo/psalm": "^4.13"
```
Comment (samsonasik) [inline composer.json]: @Ocramius should `vimeo/psalm` be in `require-dev` ? Looking at other package, it is in require-dev https://github.com/laminas/laminas-code/blob/c99ef8e5629c33bfaa3a8f1df773e916af564cd6/composer.json#L21

Comment (samsonasik) [reply composer.json]: @Ocramius I moved `vimeo/psalm` to `require-dev` https://github.com/laminas/laminas-skeleton-installer/pull/24/commits/c17ea32fafc1d5381591b075fdc5428d16ce9c99

**Comment by samsonasik** [comment]
@Ocramius All green 🎉

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Thanks @ghostwriter and @samsonasik!

**Pull request merged**